Gems/Atom/Tools/MaterialCanvas/Code/Source/Document/MaterialGraphCompiler.cpp

```diff
@@ -1122,37 +1176,80 @@ namespace MaterialCanvas
+            AZStd::string::format("Compiling %s (Started)", GetGraphPath().c_str()));
+
+        MaterialGraphCompilerNotificationBus::Event(
+            m_toolId, &MaterialGraphCompilerNotificationBus::Events::OnCompileGraphStarted, m_documentId);
     }
 
-    void MaterialGraphCompiler::CompileGraphFailed() const
+    void MaterialGraphCompiler::CompileGraphFailed()
     {
-        m_compileGraphQueued = false;
+        m_compileGraph = false;
+        m_compileAssets = false;
+        m_compileAssetIndex = 0;
         m_slotValueTable.clear();
         m_generatedFiles.clear();
-        AZ_TracePrintf_IfTrue("MaterialGraphCompiler", IsCompileLoggingEnabled(), "Compile graph failed.\n");
-        MaterialGraphCompilerNotificationBus::Event(m_toolId, &MaterialGraphCompilerNotificationBus::Events::OnCompileGraphFailed, m_documentId);
+        AZ_TracePrintf_IfTrue("MaterialGraphCompiler", IsCompileLoggingEnabled(), "Compiling failed.\n");
+
+        AtomToolsFramework::AtomToolsMainWindowRequestBus::Event(
+            m_toolId,
+            &AtomToolsFramework::AtomToolsMainWindowRequestBus::Events::SetStatusError,
+            AZStd::string::format("Compiling %s (Failed)", GetGraphPath().c_str()));
+
+        MaterialGraphCompilerNotificationBus::Event(
+            m_toolId, &MaterialGraphCompilerNotificationBus::Events::OnCompileGraphFailed, m_documentId);
     }
 
-    void MaterialGraphCompiler::CompileGraphCompleted() const
+    void MaterialGraphCompiler::CompileGraphCompleted()
     {
-        m_compileGraphQueued = false;
+        m_compileGraph = false;
+        m_compileAssets = false;
+        m_compileAssetIndex = 0;
         m_slotValueTable.clear();
-        AZ_TracePrintf_IfTrue("MaterialGraphCompiler", IsCompileLoggingEnabled(), "Compile graph completed.\n");
-        MaterialGraphCompilerNotificationBus::Event(m_toolId, &MaterialGraphCompilerNotificationBus::Events::OnCompileGraphCompleted, m_documentId);
+        AZ_TracePrintf_IfTrue("MaterialGraphCompiler", IsCompileLoggingEnabled(), "Compiling completed.\n");
+
+        AtomToolsFramework::AtomToolsMainWindowRequestBus::Event(
+            m_toolId,
+            &AtomToolsFramework::AtomToolsMainWindowRequestBus::Events::SetStatusMessage,
+            AZStd::string::format("Compiling %s (Completed)", GetGraphPath().c_str()));
+
+        MaterialGraphCompilerNotificationBus::Event(
+            m_toolId, &MaterialGraphCompilerNotificationBus::Events::OnCompileGraphCompleted, m_documentId);
     }
 
-    void MaterialGraphCompiler::BuildSlotValueTable() const
+    AZStd::string MaterialGraphCompiler::GetGraphPath() const
+    {
+        AZStd::string absolutePath;
+        AtomToolsFramework::AtomToolsDocumentRequestBus::EventResult(
+            absolutePath, m_documentId, &AtomToolsFramework::AtomToolsDocumentRequestBus::Events::GetAbsolutePath);
+
+        if (absolutePath.ends_with(".materialgraph"))
```

Comment:
Nit: Can we use `MaterialGraphExtensionWithDot` here so we don't need to duplicate the extension string?

Reply:
I had to think about that for a minute because of where I put the string definitions. This should be accessible from here but that's a weird dependency. I should find a better commissary place to move them to be accessible by all.